src/InputManager.ts

```diff
@@ -28,30 +28,14 @@ export class InputManager {
     onSend: () => void;
     session: InputSession;
 
-    inputTextArray?: Uint8Array;
-    inputMetaData?: Int32Array;
-    textEncoder: TextEncoder;
+    channel: Channel;
+    messageId = "";
 
     constructor(onSend: () => void, inputMode: InputMode) {
         this.inputMode = inputMode;
         this.session = { lineNr: 0 };
         this.batchInput = "";
-        this.textEncoder = new TextEncoder();
-        if (typeof SharedArrayBuffer !== "undefined") {
-            papyrosLog(LogType.Important, "Using SharedArrayBuffers");
-            // shared memory
-            this.inputTextArray = new Uint8Array(
-                new SharedArrayBuffer(Uint8Array.BYTES_PER_ELEMENT * 1024)
-            );
-            // 2 Int32s:
-            // index 0 indicates whether data is written
-            // index 1 denotes length of the string
-            this.inputMetaData = new Int32Array(
-                new SharedArrayBuffer(Int32Array.BYTES_PER_ELEMENT * 2)
-            );
-        } else {
-            papyrosLog(LogType.Important, "Using serviceWorker for input");
-        }
+        this.channel = makeChannel({serviceWorker: {scope: window.location.href + "/"}})!;
```

Comment:
Replace `window.location.href` with `serviceWorkerRoot` or `inputURL` or whatever from your most recent commits. `scope` should be a URL that is *within* the scope of the registered service worker, so that sync-message can make requests to `scope` plus an extra path and know that they will be intercepted.

Reply:
I'm merging this myself to ensure it all works properly within Dodona itself. For ease of use, the location of the service worker should be configurable while still allowing the worker to do all its tasks.